numpy/testing/utils.py

```diff
@@ -1178,9 +1178,11 @@ def assert_raises(*args,**kwargs):
     return nose.tools.assert_raises(*args,**kwargs)
 
 
-def assert_raises_regex(exception_class, expected_regexp,
-                        callable_obj=None, *args, **kwargs):
+def assert_raises_regex(exception_class, expected_regexp, *args, **kwargs):
     """
+    assert_raises_regex(exception_class, expected_regexp, callable_obj=None,
```

Comment:
Hmm, checking shows that unittest has
```
    def assertRaisesRegex(self, expected_exception, expected_regex,
                          *args, **kwargs):
```
e.g., no mention of callable, although the official documentation includes it as a purely positional argument. I wonder if we should just omit the `=None` here in the documentation and mention that when used as a context manager only the first two arguments are needed. Or maybe go all long winded and just keep the first two arguments and document the parameters with `callable_obj` documented as being omitted when used as context manager. The Python 3.5 documentation has
```
            expected_exception: Exception class expected to be raised.
            expected_regex: Regex (re pattern object or string) expected
                    to be found in error message.
            args: Function to be called and extra positional args.
            kwargs: Extra kwargs.
            msg: Optional message used in case of failure. Can only be used
                    when assertRaisesRegex is used as a context manager.
```
with callable omitted from the signature. Unfortunately, `msg` is not available in 2.7.